### PR TITLE
feat(inventory): display more data from 'locked items'

### PR DESCRIPTION
**Pull request by stonebuzz**
Display more data from 'locked items' (link if possible, field if possible etc ..)

image

| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

### CODE REVIEW

**Review by orthagh — APPROVED** [review]

File: src/Lock.php [hunk 1]
```diff
@@ -302,20 +308,28 @@ public static function showForItem(CommonDBTM $item)
             $first  = true;
             foreach ($DB->request($item_disk->getTable(), $params) as $line) {
                 if ($first) {
-                    echo "<tr><th colspan='2'>" . $item_disk->getTypeName(Session::getPluralNumber()) . "</th></tr>\n";
+                    echo "<tr>";
+                    echo "<th width='10'></th>";
+                    echo "<th>" . $item_disk->getTypeName(Session::getPluralNumber()) . "</th>";
+                    echo "<th>" . __('Partitiont') . "</th>";
```
Comment (cedric-anne) [inline src/Lock.php:314]:
```suggestion
                    echo "<th>" . __('Partition') . "</th>";
```

File: src/Lock.php [hunk 2]
```diff
@@ -381,21 +423,36 @@ public static function showForItem(CommonDBTM $item)
...
                 echo "<input type='checkbox' name='Item_SoftwareVersion[" . $data['id'] . "]'>";
             }
-            echo "</td>";
 
-            echo "<td class='left' width='95%'>" . $data['software'] . " " . $data['version'] . "</td>";
-            echo "</tr>\n";
+            $sversion = new SoftwareVersion();
+            $sversion->getFromDB($item_sv->fields['softwareversions_id']);
+
+            $softawre = new Software();
```
Comment (cedric-anne) [inline src/Lock.php:447]:
```suggestion
            $software = new Software();
```

File: src/Lock.php [hunk 3]
```diff
@@ -381,21 +423,36 @@ public static function showForItem(CommonDBTM $item)
...
+            $softawre->getFromDB($sversion->fields['softwares_id']);
+
+            echo "</td>";
+            echo "<td class='left'>" . $softawre->fields['name'] . "</td>"; //no form for item software version
+            echo "<td class='left'>" . $sversion->fields['name'] . "</td>";
+            echo "<td class='left'>" . $item_sv->fields['date_install'] . "</td>";
```
Comment (cedric-anne) [inline src/Lock.php:453]:
```suggestion
            echo "<td class='left'>" . Html::convDateTime($item_sv->fields['date_install']) . "</td>";
```

File: src/Lock.php [hunk 4]
```diff
@@ -492,20 +581,33 @@ public static function showForItem(CommonDBTM $item)
...
+            echo "<td class='left'>" . $networkname->getLink() . "</td>";
 
-            echo "<td class='left' width='95%'>" . $networkname->getName() . "</td>";
+            $fqdn = new FQDN();
+            $fqdn_name = "";
+            if ($fqdn->getFromDB($networkname->fields['fqdns_id'])) {
+                $fqdn_name = $fqdn->fields['name'];
+            }
+            echo "<td class='left'></td>";
+            echo "<td class='left'>" . $fqdn_name . "</td>";
```
Comment (cedric-anne) [inline src/Lock.php:609]: Header is contained in the 3rd `th`.
```suggestion
            echo "<td class='left'>" . $fqdn_name . "</td>";
            echo "<td class='left'></td>";
```

File: src/Lock.php [hunk 5]
```diff
@@ -530,20 +632,27 @@ public static function showForItem(CommonDBTM $item)
...
                 $first = false;
             }
 
             echo "<tr class='tab_bg_1'>";
-
             echo "<td class='center' width='10'>";
             if ($ipaddress->can($line['id'], UPDATE) || $ipaddress->can($line['id'], PURGE)) {
                 $header = true;
-                echo "<input type='checkbox' name='IPAddress[" . $line['id'] . "]'>";
+                echo "<input type='checkbox' name='Item_Disk[" . $line['id'] . "]'>";
```
Comment (cedric-anne) [inline src/Lock.php:649]: Are you sure of this change ?
```suggestion
                echo "<input type='checkbox' name='IPAddress[" . $line['id'] . "]'>";
```

File: src/Lock.php [hunk 6]
```diff
@@ -619,7 +740,7 @@ public static function showForItem(CommonDBTM $item)
             ]);
         } else {
             echo "<tr class='tab_bg_2'>";
-            echo "<td class='center' colspan='2'>" . __('No locked item') . "</td></tr>";
+            echo "<td class='center' colspan='4'>" . __('No locked item') . "</td></tr>";
```
Comment (cedric-anne) [inline src/Lock.php:743]:
```suggestion
            echo "<td class='center' colspan='5'>" . __('No locked item') . "</td></tr>";
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

File: src/Lock.php (hunk 2 above)
Comment (stonebuzz) [reply src/Lock.php]: fixed

File: src/Lock.php (hunk 3 above)
Comment (stonebuzz) [reply src/Lock.php]: fixed

File: src/Lock.php (hunk 4 above)
Comment (stonebuzz) [reply src/Lock.php]: fixed

File: src/Lock.php (hunk 5 above)
Comment (stonebuzz) [reply src/Lock.php]: reverted to "IPAddress"

File: src/Lock.php (hunk 6 above)
Comment (stonebuzz) [reply src/Lock.php]: fixed

**Review by cedric-anne — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**